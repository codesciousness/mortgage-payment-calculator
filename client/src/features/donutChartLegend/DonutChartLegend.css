.DonutChartLegend {
    width: 100%;
    margin: 1rem;
    padding: 1rem;
}

.DonutChartLegend__container {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.DonutChartLegend__label, .DonutChartLegend__amount {
    margin: 1rem 0;
}

.color__codes {
    position: relative;
    left: 23px;
}

.color__codes:before {
    content: '';
    position: absolute;
    left: -23px;
    top: 3px;
    width: 16px;
    height: 16px;
    border-radius: 50%;
}

.color__codes.principal:before {
    background-color: DeepSkyBlue;
}

.color__codes.propertyTax:before {
    background-color: LightSkyBlue;
}

.color__codes.homeownersInsurance:before {
    background-color: SkyBlue;
}

.color__codes.PMI:before {
    background-color: PowderBlue;
}

.color__codes.HOAFees:before {
    background-color: PaleTurquoise;
}

@media only screen and (max-width: 425px) {

    .DonutChartLegend {
        padding: 1rem 0 0 0;
    }
}

@media only screen and (max-width: 375px) {

    .DonutChartLegend__container {
        flex-direction: column;
        align-items: flex-start;
    }

    .DonutChartLegend__label, .DonutChartLegend__amount {
        margin: 0.25rem 1rem;
    }

    .DonutChartLegend__amount {
        margin-left: 2.3rem;
    }
}