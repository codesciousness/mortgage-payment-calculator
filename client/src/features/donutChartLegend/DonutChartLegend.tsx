import React from 'react';
import Box from '@mui/material/Box';
import './DonutChartLegend.css';
import { selectPropertyTaxes, selectHomeInsurance, selectHOAFees, selectOtherCosts, 
    selectMortgagePayment, selectMonthlyPayment } from '../loansSlice';
import { useAppSelector } from '../../app/hooks';

const styles = {
    display: 'flex', 
    justifyContent: 'space-between', 
    alignItems: 'center', 
    borderBottom: 1, 
    borderColor: 'divider'
}

const DonutChartLegend = (): JSX.Element => {
    const propertyTaxes = useAppSelector(selectPropertyTaxes);
    const homeInsurance = useAppSelector(selectHomeInsurance);
    const hoaFees = useAppSelector(selectHOAFees);
    const otherCosts = useAppSelector(selectOtherCosts);
    const mortgagePayment = useAppSelector(selectMortgagePayment);
    const monthlyPayment = useAppSelector(selectMonthlyPayment);

    return (
        <section id='DonutChartLegend' className='DonutChartLegend'>
            <Box sx={styles}>
                <p className='DonutChartLegend__label color__codes principal'>Principal &amp; interest</p>
                <p className='DonutChartLegend__amount'>${mortgagePayment}</p>
            </Box>
            <Box sx={styles}>
                <p className='DonutChartLegend__label color__codes propertyTax'>Property tax</p>
                <p className='DonutChartLegend__amount'>${propertyTaxes.dollar}</p>
            </Box>
            <Box sx={styles}>
                <p className='DonutChartLegend__label color__codes homeownersInsurance'>Homeowner's insurance</p>
                <p className='DonutChartLegend__amount'>${homeInsurance.dollar}</p>
            </Box>
            <Box sx={styles}>
                <p className='DonutChartLegend__label color__codes HOAFees'>HOA fees</p>
                <p className='DonutChartLegend__amount'>${hoaFees.dollar}</p>
            </Box>
            <Box sx={styles}>
                <p className='DonutChartLegend__label color__codes otherCosts'>Other costs</p>
                <p className='DonutChartLegend__amount'>${otherCosts.dollar}</p>
            </Box>
            <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
                <p className='DonutChartLegend__label totalMonthlyPayment'><b>Total monthly payment</b></p>
                <p className='DonutChartLegend__amount'><b>${monthlyPayment}</b></p>
            </Box>
        </section>
    );
};

export default DonutChartLegend;