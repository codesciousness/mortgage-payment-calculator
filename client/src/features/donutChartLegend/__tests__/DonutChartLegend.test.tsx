import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { store } from '../../../app/store';
import DonutChartLegend from '../DonutChartLegend';

describe('<DonutChartLegend />', () => {
    test('DonutChartLegend component is defined', () => {
        expect(<DonutChartLegend/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(
            <Provider store={store}>
                <DonutChartLegend/>
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly with initial state from Redux store', () => {
        render(
            <Provider store={store}>
                <DonutChartLegend/>
            </Provider>
        );
        expect(screen.getByText('Principal & interest')).toBeInTheDocument();
        expect(screen.getByText('$1,838')).toBeInTheDocument();
        expect(screen.getByText('Property tax')).toBeInTheDocument();
        expect(screen.getByText('$315')).toBeInTheDocument();
        expect(screen.getByText("Homeowner's insurance")).toBeInTheDocument();
        expect(screen.getByText('$175')).toBeInTheDocument();
        expect(screen.getByText('Private mortgage insurance')).toBeInTheDocument();
        expect(screen.getByText('HOA fees')).toBeInTheDocument();
        expect(screen.getByText('Total monthly payment')).toBeInTheDocument();
        expect(screen.getByText('$2,328')).toBeInTheDocument();
    });
});