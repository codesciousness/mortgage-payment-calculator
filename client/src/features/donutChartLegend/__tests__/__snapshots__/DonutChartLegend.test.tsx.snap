// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DonutChartLegend /> component matches the snapshot 1`] = `
<section
  className="DonutChartLegend"
  id="DonutChartLegend"
>
  <div
    className="DonutChartLegend__container MuiBox-root css-1gsv261"
  >
    <p
      className="DonutChartLegend__label color__codes principal"
    >
      Principal & interest
    </p>
    <p
      className="DonutChartLegend__amount"
    >
      $
      1,838
    </p>
  </div>
  <div
    className="DonutChartLegend__container MuiBox-root css-1gsv261"
  >
    <p
      className="DonutChartLegend__label color__codes propertyTax"
    >
      Property tax
    </p>
    <p
      className="DonutChartLegend__amount"
    >
      $
      315
    </p>
  </div>
  <div
    className="DonutChartLegend__container MuiBox-root css-1gsv261"
  >
    <p
      className="DonutChartLegend__label color__codes homeownersInsurance"
    >
      Homeowner's insurance
    </p>
    <p
      className="DonutChartLegend__amount"
    >
      $
      175
    </p>
  </div>
  <div
    className="DonutChartLegend__container MuiBox-root css-1gsv261"
  >
    <p
      className="DonutChartLegend__label color__codes PMI"
    >
      Private mortgage insurance
    </p>
    <p
      className="DonutChartLegend__amount"
    >
      $
      0
    </p>
  </div>
  <div
    className="DonutChartLegend__container MuiBox-root css-1gsv261"
  >
    <p
      className="DonutChartLegend__label color__codes HOAFees"
    >
      HOA fees
    </p>
    <p
      className="DonutChartLegend__amount"
    >
      $
      0
    </p>
  </div>
  <div
    className="DonutChartLegend__container MuiBox-root css-0"
  >
    <p
      className="DonutChartLegend__label totalMonthlyPayment"
    >
      <b>
        Total monthly payment
      </b>
    </p>
    <p
      className="DonutChartLegend__amount"
    >
      <b>
        $
        2,328
      </b>
    </p>
  </div>
</section>
`;
